editoast/src/infra.rs

```diff
@@ -1,21 +1,24 @@
 use crate::error::Result;
 use crate::generated_data;
 use crate::infra_cache::InfraCache;
+use crate::models::Identifiable;
 use crate::tables::osrd_infra_infra;
 use crate::tables::osrd_infra_infra::dsl;
 use chrono::{NaiveDateTime, Utc};
 use diesel::result::Error as DieselError;
 use diesel::sql_types::{BigInt, Bool, Nullable, Text};
 use diesel::ExpressionMethods;
-use diesel::{delete, sql_query, update, PgConnection, QueryDsl, RunQueryDsl};
-use editoast_derive::EditoastError;
+use diesel::{sql_query, update, PgConnection, QueryDsl, RunQueryDsl};
+use editoast_derive::{EditoastError, Model};
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
 pub const RAILJSON_VERSION: &str = "3.2.0";
 
-#[derive(Clone, QueryableByName, Queryable, Debug, Serialize, Deserialize, Identifiable)]
+#[derive(Clone, QueryableByName, Queryable, Debug, Serialize, Deserialize, Identifiable, Model)]
 #[diesel(table_name = osrd_infra_infra)]
+#[model(table = "osrd_infra_infra")]
+#[model(delete)]
```

Comment:
This is done and will conflict with #3718.
Are you okay to rebase and merge this PR after that?